tvb/simulator/integrators.py

```diff
@@ -177,13 +198,19 @@ def scheme(self, X, dfun, coupling, local_coupling, stimulus):
         """
         #import pdb; pdb.set_trace()
         m_dx_tn = dfun(X, coupling, local_coupling)
-        inter = X + self.dt * (m_dx_tn  + stimulus)
-        self.clamp_state(inter)
+        inter = X + self.dt * (m_dx_tn + stimulus)
+        if self.constraint_state_variable_boundaries is not None:
```

Comment:
this is a lot of boilerplate, should be refactored into the call site in the simulator